### DISCUSSION TITLE
got dependency error when use docker-compose build

### DISCUSSION THREAD

**Discussion by delong1998** [requarks/wiki#4292] [Error / Bug Report]
first i was forked the project in my repositories and then built it as docker image to dockerhub
finally, got  a dependency error  when i using docker-compose to build the project ,and this is my docker-compose file
```
version: "3"
services:
  db:
    image: postgres:11-alpine
    environment:
      POSTGRES_DB: wiki
      POSTGRES_PASSWORD: wikijsrocks
      POSTGRES_USER: wikijs
    logging:
      driver: "none"
    restart: unless-stopped
    volumes:
      - wiki-database:/var/lib/postgresql/data

  wiki:
    image: delong1998/ukylin:dev
    depends_on:
      - db
    environment:
      DB_TYPE: postgres
      DB_HOST: db
      DB_PORT: 5432
      DB_USER: wikijs
      DB_PASS: wikijsrocks
      DB_NAME: wiki
    restart: unless-stopped
    volumes:
      - wiki-data:/app/data
      - wiki-backup:/backup
    ports:
      - "10006:3000"

volumes:
  wiki-data:
    driver: local
    driver_opts:
      type: none
      device: ./data
      o: bind
  wiki-backup:
    driver: local
    driver_opts:
      type: none
      device: ./backup
      o: bind
  wiki-database:
    
 
```
got a problem when i run docker-compose

```
2021-08-03T05:50:36.956Z [MASTER] info: =======================================
2021-08-03T05:50:36.957Z [MASTER] info: = Wiki.js 2.0.0 =======================
2021-08-03T05:50:36.957Z [MASTER] info: =======================================
2021-08-03T05:50:36.957Z [MASTER] info: Initializing...
2021-08-03T05:50:37.312Z [MASTER] info: Using database driver pg for postgres [ OK ]
2021-08-03T05:50:37.315Z [MASTER] info: Connecting to database...
2021-08-03T05:50:37.328Z [MASTER] info: Database Connection Successful [ OK ]
2021-08-03T05:50:37.677Z [MASTER] warn: Mail is not setup! Please set the configuration in the administration area!
2021-08-03T05:50:37.679Z [MASTER] error: Cannot find module 'graphql-upload'
Require stack:
- /wiki/server/middlewares/upload.js
- /wiki/node_modules/auto-load/lib/index.js
- /wiki/server/core/db.js
- /wiki/server/core/kernel.js
- /wiki/server/index.js
```

**Comment by craph** [comment]
Hello @delong1998 first of all, did your forked repo is up-to-date from the upstream ?

Then, refering to the documentation about dev, when you want to build for production you need to build assets first and then the docker image. Here is the link : https://docs.requarks.io/dev#building-production-assets and https://docs.requarks.io/dev#build-production-images

**Comment by bruot** [comment]
Hi. I don't know if this is related, but I have a similar issue with the `requarks/wiki:canary-2.5.206` image on Docker hub. With a fresh DB container and my docker-compose.yml that works with version 204, I also get `Cannot find module 'graphql-upload'` errors and the web server does not start after setting up the admin account.

**Comment by delong1998** [comment]
HI, @craph thanks for your replay, it gives me a lot of inspiration. and The problem has been solved
here's my solution,
i find there is no "graphql-upload"  dependencies in package.json,  so i  add the  dependency  to package.json
`"graphql-upload": "12.0.0"`
@bruot  and hope it worked for you
